test/parallel/test-http-server-optimize-empty-requests.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+const net = require('net');
+
+let reqs = 0;
+let optimizedReqs = 0;
+const server = http.createServer({
+  optimizeEmptyRequests: true
+}, (req, res) => {
+  reqs++;
+  if (req._dumped) {
+    optimizedReqs++;
+    req.on('data', common.mustNotCall());
+    req.on('end', common.mustNotCall());
+
+    assert.strictEqual(req._dumped, true);
+    assert.strictEqual(req.readableEnded, true);
+    assert.strictEqual(req.destroyed, true);
+  }
+  res.writeHead(200);
+  res.end('ok');
+});
```

Comment:
nit: it's not clear in this test what "be optimized" means. Expanding on some code comments could help provide context to anyone coming into this code later.